core/src/main/java/org/elasticsearch/index/query/FilteredQueryBuilder.java

```diff
@@ -40,6 +46,15 @@
     static final FilteredQueryBuilder PROTOTYPE = new FilteredQueryBuilder(null, null);
 
     /**
+     * A query that applies a filter to the results of a match_all query.
+     * @param filterBuilder The filter to apply on the query (Can be null)
+     * */
+    public FilteredQueryBuilder(@Nullable QueryBuilder filterBuilder) {
+        this.queryBuilder = new MatchAllQueryBuilder();
+        this.filterBuilder = filterBuilder;
+    }
+
```

Comment:
I just looked at the original parser code and the docs again and saw that it explicitly allows leaving out the `filter` section:
```
// we allow for null filter, so it makes compositions on the client side to be simpler
```
The JSON side of this query seems to be currently seems super lenient (basically both filter/query can be null), all null-values map to some defaults. If we restrict this on the JAVA-API, we should maybe be explicit about it in this case, saying which nested queries to use to mimic the JSON behaviour when leaving the clauses there.
I'm on the fence about adding null-checks in constructors after reading the original parser code, maybe we should also allow it in the Java-API but internally always set the defaults to avoid null-checks all over the place. Sorry for this back & forth.